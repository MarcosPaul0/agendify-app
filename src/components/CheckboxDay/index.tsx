import { Text, TouchableOpacity } from 'react-native';
import { Checkbox } from '@components/Checkbox';
import { ICheckboxDayProps } from './interfaces/checkboxDayProps.interface';

export function CheckboxDay({ day, isChecked, onCheck }: ICheckboxDayProps) {
  const dayBackground = isChecked ? 'bg-BLUE_100' : 'bg-GRAY_100';

  return (
    <TouchableOpacity
      testID={day}
      className={`
        px-8 py-3.5 flex-row items-center
        justify-between ${dayBackground}
      `}
      onPress={onCheck}
    >
      <Text className="font-bold text-base text-BLUE_800">{day}</Text>
      <Checkbox isChecked={isChecked} />
    </TouchableOpacity>
  );
}
