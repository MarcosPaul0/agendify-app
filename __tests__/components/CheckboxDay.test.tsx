import { render } from '@testing-library/react-native';
import { CheckboxDay } from '@components/CheckboxDay';

describe('Checkbox Day', () => {
  it('Match with snapshot', async () => {
    const tree = render(
      <CheckboxDay day="fake-day" isChecked onCheck={() => null} />
    );

    expect(tree).toMatchSnapshot();
  });
});
