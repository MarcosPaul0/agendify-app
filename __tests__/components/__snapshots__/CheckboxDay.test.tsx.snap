// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Day Match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#C1D1EA",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 14,
    }
  }
  testID="fake-day"
>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#18273C",
        },
      ]
    }
  >
    fake-day
  </Text>
  <View
    style={
      [
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#3763A4",
          "borderLeftColor": "#3763A4",
          "borderRightColor": "#3763A4",
          "borderTopColor": "#3763A4",
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        {
          "backgroundColor": "#3763A4",
        },
        {
          "height": 24,
        },
        {
          "width": 24,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      color="#F8F9FA"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={20}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            undefined,
            {},
          ],
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      tintColor="#F8F9FA"
      vbHeight={256}
      vbWidth={256}
      width={20}
    >
      <RNSVGGroup>
        <RNSVGRect
          fill={null}
          height="256"
          propList={
            [
              "fill",
            ]
          }
          width="256"
          x="0"
          y="0"
        />
        <RNSVGPath
          d="M216 72 104 184 48 128"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294507002,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="16"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
